pyfar/utils.py

```diff
@@ -127,3 +127,80 @@ def broadcast_cdims(signals, cdim=None):
     if cdim is None:
         cdim = np.max([len(s.cshape) for s in signals])
     return [broadcast_cdim(s, cdim) for s in signals]
+
+
+def concatenate(signals, caxis=0, broadcasting=False):
+    """
+    Merge multiple Signal, Timedata or Frequencydata objects along a given
+    caxis.
+
+    Parameters
+    ----------
+    signals : tuple of Signal, TimeData or FrequencyData
+        The signals to concatenate. All signals must be of the same object type
+        and either have the same cshape or be broadcastable to the same cshape,
+        except in the dimension corresponding to caxis (the first, by default).
+        If this is the case, set ``broadcasting=True``.
+    caxis : int
+        The caxis along which the signals are concatenated. More details and
+        background about caxis is given in the concepts of
+        :py:mod:`Audio classes <pyfar._concepts.audio_classes>`.
+        The default is ``0``.
+    broadcasting: bool
+        If this is ``True``, the signals will be broadcasted to common
+        cshape, except for the caxis along which the signals are
+        concatenated.
+        The :py:mod:`cshape <pyfar._concepts.audio_classes>` of the signals are
+        broadcasted following the `numpy broadcasting rules
+        <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+        The default is ``False``.
+    Returns
+    -------
+    merged : Signal
+        The merged signal object.
+    """
+    # check input
+    for signal in signals:
+        if not isinstance(signal, (pf.Signal, pf.TimeData, pf.FrequencyData)):
+            raise TypeError("All input data must be of type pf.Signal, "
+                            "pf.TimeData or pf.FrequencyData.")
+    if not isinstance(broadcasting, bool):
+        raise TypeError("'broadcasting' needs to be False or True.")
+    # check matching meta data of input signals.
+    [signals[0]._assert_matching_meta_data(s) for s in signals]
```

Comment:
Just a tiny note - I think you can pass all audio objects at the same time
```suggestion
    _assert_matching_meta_data(signals)
```